mongo/database.go

```diff
@@ -507,6 +507,156 @@ func (db *Database) Watch(ctx context.Context, pipeline interface{},
 //
 // For more information about the command, see https://docs.mongodb.com/manual/reference/command/create/.
 func (db *Database) CreateCollection(ctx context.Context, name string, opts ...*options.CreateCollectionOptions) error {
+	cco := options.MergeCreateCollectionOptions(opts...)
+	ef := cco.EncryptedFields
+	// Follow Client-Side Encryption specification to check for encryptedFields.
+	// Check for encryptedFields from create options.
+	if ef == nil {
+		ef = db.getEncryptedFieldsFromMap(name)
+	}
+	// Check for encryptedFields from the client EncryptedFieldsMap.
```

Comment:
I think this comment should precede the previous `if ef == nil` check. The comments on (current) L512 and L513 should probably precede `ef := cco.EncryptedFields`.